### ISSUE TITLE
[0.16.2 regression]  `{"key": null}` fails to validate as `{"type": "object"}`

### ISSUE THREAD

**Issue by andersk** [python-openapi/openapi-core#442]
In 0.16.1, `{"key": null}` validated successfully against `{"type": "object"}`—an object that includes a `null` value is still an object. But in 0.16.2 this incorrectly raises `openapi_core.unmarshalling.schemas.exceptions.InvalidSchemaValue: Value None not valid for schema of type any: (<ValidationError: 'None for not nullable'>,)`.

This regression was introduced by commit 692a9156abd99a63cfbaf018bf26470d0060b702 (#434).

Full reproducible example:

```python
from openapi_core import Spec
from openapi_core.testing import MockRequest
from openapi_core.validation.request import openapi_request_validator

spec = Spec.create(
    {
        "openapi": "3.0.3",
        "info": {"title": "test", "version": "0"},
        "paths": {
            "/test": {
                "post": {
                    "parameters": [
                        {
                            "name": "obj",
                            "in": "query",
                            "content": {
                                "application/json": {
                                    "schema": {"type": "object"},
                                }
                            },
                        },
                    ],
                    "responses": {"200": {"description": ""}},
                },
            },
        },
    },
)
request = MockRequest(
    "http://localhost/", "post", "/test", args={"obj": '{"key": null}'}
)
result = openapi_request_validator.validate(spec, request)
print(result)
result.raise_for_errors()
```

**Comment by Wim-De-Clercq** [comment]
The openapi-schema-validator update introduced this https://github.com/p1c2u/openapi-schema-validator/blob/master/openapi_schema_validator/validators.py#L62
Which makes every schema have a default "nullable: false" requirement. Which I believe is correct. You should specify "nullable: true" if you want to have nullable properties.

The problem is that openapi-core creates a little dummy schema for the "additionalProperties" -- for any key not present in the schema -- like so:
https://github.com/p1c2u/openapi-core/blob/4055a365591d9ef5ac8c93a5ebaee61c19f04fe0/openapi_core/unmarshalling/schemas/unmarshallers.py#L295

But now that little schema will have a "nullable: false" check, and it incorrectly raises exceptions for this.

The above line should be updated to
```py
additional_prop_schema = Spec.from_dict({"nullable": True})
```

**Comment by p1c2u** [comment]
> Or perhaps a little refactor is in place to not create any schema and just take the value as is. It feels a bit wasteful to constantly create these little schemas and do null-validation checks for something which isn't even defined in the schema in the first place.

I must admit it was silly. I will change that .

**Comment by Wim-De-Clercq** [comment]
@p1c2u 
Do you reckon this bug is actually fixed in the meantime? If I git-checkout 0.17.1 and run the above test (with updated imports to reflect changes) it seems to report no errors.

**Comment by p1c2u** [comment]
Yes it was merged hence closing
